x/cash/controller.go

```diff
@@ -55,7 +55,7 @@ func (c BaseController) MoveCoins(store weave.KVStore,
 	src weave.Address, dest weave.Address, amount x.Coin) error {
 
 	if !amount.IsPositive() {
-		return ErrInvalidAmount("Non-positive SendMsg")
+		return errors.ErrInvalidAmount.Newf("non-positive SendMsg: %#v", &amount)
```

Comment:
Nice.

This `%#v` format is proper, but a bit verbose.
Maybe nicer to add a `String()` method to `x.Coin` for nice human-readable errors.
(This is optional: in this PR, later, never...)

Reply:
so the %#v is what actually uses the fmt stringer when available, hence it’s used for all the fmt.Stringer satisfying objects. 

I wanted to add News (for stringer) but deemed it too much and %#v should cover both stringer and value when not available

Reply:
Yeah, `%#v` includes the field names of the struct, which `%v` omits. 
And then `%+v` with stacktrace if you use pkg/errors

But fine for now, this is a bigger change